java_src/foam/nanos/boot/NSpecFactory.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.boot;
+
+import foam.core.*;
+import foam.nanos.*;
+
+public class NSpecFactory implements XFactory {
+  NSpec spec_;
+
+  public NSpecFactory(NSpec spec) {
+    spec_ = spec;
+  }
+
+  public Object create(X x) {
+    NanoService ns = null;
+
+    try {
+      ns = spec_.createService();
+
+      ((ContextAwareSupport) ns).setX(x);
```

Comment:
We should make NanoService extend ContextAware (waiting for bug fix by @marcogbarcellos), then this cast won't be needed.